numpy/core/function_base.py

```diff
@@ -140,7 +140,7 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
         dtype = dt
 
     delta = stop - start
```

Comment:
Ideally you'd do this computation while forcing the result to be an array and not a scalar, which would work better for nested object arrays. Perhaps ufuncs should gain a `keeparray` argument for this...

Reply:
It may be simpler to change the current behaviour of ufuncs: array scalar plus array scalar should not decay to a regular scalar.

Reply:
Isn't the problem in line 135 and 136.

```
In [21]: type(np.asanyarray(np.array(1))*1.0)
Out[21]: numpy.float64

In [22]: type(np.asanyarray(np.array(1, dtype='O'))*1.0)
Out[22]: float
```

Reply:
@jenshnielsen - indeed, you are right. the problem is earlier, though it is the same issue: that an array scalar is allowed to decay to a regular one.